File: src/packets/PacketS2CZoneIndex.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x03
 * Zone Index
 * Informs client of all the existing zones when it connects
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 */
class PacketS2CZoneIndex extends PacketS2C {
    constructor(zones) {
        super();
        for (let i = 0; i < zones.length; i++) {
            this._buffer.writeUInt8(zones[i][0]);
            this._buffer.writeUInt8(zones[i][1]);
        }
    }

    getID() {
        return 0x03;
    }
}

module.exports = PacketS2CZoneIndex;

File: src/RNetPi.Core/Packets/PacketS2CZoneIndex.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x03
/// Zone Index
/// Informs client of all the existing zones when it connects
/// Data:
///     (Array of) (Unsigned Char) Controller ID
///     (Array of) (Unsigned Char) Zone ID
/// </summary>
public class PacketS2CZoneIndex : PacketS2C
{
    public const byte ID = 0x03;

    public PacketS2CZoneIndex(IEnumerable<(byte ControllerID, byte ZoneID)> zones)
    {
        foreach (var (controllerID, zoneID) in zones)
        {
            Writer.Write(controllerID);
            Writer.Write(zoneID);
        }
    }

    public override byte GetID() => ID;
}